lib/semantics/check-declarations.cpp

```diff
@@ -38,7 +38,7 @@ class CheckHelper {
   }
   void Check(const ArraySpec &);
   void Check(const DeclTypeSpec &, bool canHaveAssumedTypeParameters);
-  void Check(const Symbol &);
+  void Check(Symbol &);
```

Comment:
Is there an unavoidable reason for removing the `const` qualification on these references?

Reply:
I was able to avoid the multiple messages on the same line by distinguishing between polymorphic types and assumed-type types.  This let me restore the `const`ness of the `Symbol` arguments.  I pushed a commit with these changes.

Reply:
OK.  Changes for all other comments have already been pushed.  Let me know if want any other changes.